ingest/tests/common/mod.rs

```diff
@@ -0,0 +1,134 @@
+use anyhow::bail;
+use backon::{ExponentialBuilder, Retryable};
+use file_store::file_sink::FileSinkClient;
+use helium_crypto::{KeyTag, Keypair, Network, Sign};
+use helium_proto::services::poc_mobile::{
+    Client as PocMobileClient, VerifiedSubscriberMappingEventReqV1,
+    VerifiedSubscriberMappingEventResV1,
+};
+use ingest::server_mobile::GrpcServer;
+use prost::Message;
+use rand::rngs::OsRng;
+use std::{net::SocketAddr, sync::Arc};
+use tokio::{net::TcpListener, sync::mpsc::Receiver};
+use tonic::{
+    metadata::{Ascii, MetadataValue},
+    transport::Channel,
+    Request,
+};
+use triggered::Trigger;
+
+pub async fn setup_mobile() -> anyhow::Result<(TestClient, Trigger)> {
+    let key_pair = generate_keypair();
+
+    let (file_sink_tx, file_sink_rx) = tokio::sync::mpsc::channel(10);
+    let file_sink = FileSinkClient::new(file_sink_tx, "test_file_sync");
+
+    let socket_addr = {
+        let tcp_listener = TcpListener::bind("127.0.0.1:0").await?;
+        tcp_listener.local_addr()?
+    };
+
+    let token = "api_token";
+    let api_token = format!("Bearer {token}")
+        .parse::<MetadataValue<_>>()
+        .ok()
+        .unwrap();
+
+    let (trigger, listener) = triggered::trigger();
+
+    tokio::spawn(async move {
+        let grpc_server = GrpcServer::new(
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            file_sink.clone(),
+            Network::MainNet,
+            socket_addr,
+            api_token,
+        );
+
+        grpc_server.run(listener).await
+    });
+
+    let client = TestClient::new(socket_addr, key_pair, token.to_string(), file_sink_rx).await;
+
+    Ok((client, trigger))
+}
+
+pub struct TestClient {
+    client: PocMobileClient<Channel>,
+    key_pair: Arc<Keypair>,
+    authorization: MetadataValue<Ascii>,
+    file_sink_rx: Receiver<file_store::file_sink::Message>,
+}
+
+impl TestClient {
+    pub async fn new(
+        socket_addr: SocketAddr,
+        key_pair: Keypair,
+        api_token: String,
+        file_sink_rx: Receiver<file_store::file_sink::Message>,
+    ) -> TestClient {
+        let client = (|| PocMobileClient::connect(format!("http://{socket_addr}")))
+            .retry(&ExponentialBuilder::default())
+            .await
+            .expect("client connect");
+
+        TestClient {
+            client,
+            key_pair: Arc::new(key_pair),
+            authorization: format!("Bearer {}", api_token).try_into().unwrap(),
+            file_sink_rx,
+        }
+    }
+
+    pub async fn recv(mut self) -> anyhow::Result<Vec<u8>> {
+        match self.file_sink_rx.recv().await {
```

Comment:
May want to add a timeout so the test cannot hang indefinitely if something goes wrong. `mobile_verifier/tests/integrations/common/mod.rs` has an a example.